lib/srfi/1/misc.scm

```diff
@@ -15,12 +15,13 @@
 (define (concatenate! lists)
   (if (null? lists)
       '()
-      (let lp ((ls lists))
-        (cond ((not (pair? (cdr ls)))
-               (car lists))
-              (else
-               (set-cdr! (last-pair (car ls)) (cadr ls))
-               (lp (cdr ls)))))))
+      (fold (lambda (el acc)
```

Comment:
Can you add a unit test for this case?

Reply:
Added in 70989e0cef3ccb83d7391582c1d0ab0459bb1954